### PR TITLE
libc/arm: add support of PACBTI

### PR DESCRIPTION
**Pull request by anchao**
## Summary

libc/arm: add support of PACBTI

Reference:
https://developer.arm.com/documentation/100748/0617/Security-features-supported-in-Arm-Compiler-for-Embedded/PACBTI-M-extension-mitigations-against-ROP-and-JOP-style-attacks
https://developer.arm.com/documentation/101754/0619/armclang-Reference/armclang-Command-line-Options/-mbranch-protection

Signed-off-by: chao an <anchao@xiaomi.com>

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
please rebase to the last master which fix the recent ci break

File: libs/libc/machine/arm/armv8-m/gnu/arch_strlen.S [hunk 1]
```diff
@@ -169,6 +184,9 @@ def_fn	strlen p2align=6
 	clz	data1a, data1a
 	ldrd	r4, r5, [sp], #8
 	add	result, result, data1a, lsr #3	/* Bits -> Bytes.  */
+#if __ARM_FEATURE_PAC_DEFAULT
+	aut ip, lr, sp
```
Comment (xiaoxiang781216) [inline libs/libc/machine/arm/armv8-m/gnu/arch_strlen.S:188]: align `ip` with other line

File: libs/libc/machine/arm/armv8-m/gnu/arch_strlen.S [hunk 2]
```diff
@@ -108,6 +116,13 @@
 #define tmp2		r5
 
 def_fn	strlen p2align=6
+#if __ARM_FEATURE_PAC_DEFAULT
+#  if __ARM_FEATURE_BTI_DEFAULT
+	pacbti  ip, lr, sp
+#  else
+	pac ip, lr, sp
```
Comment (xiaoxiang781216) [inline libs/libc/machine/arm/armv8-m/gnu/arch_strlen.S:123]: ditto

File: libs/libc/machine/arm/armv8-m/gnu/arch_strcpy.S [hunk 3]
```diff
@@ -209,6 +224,9 @@ strcpy:
     mov     tmp3, #0
     strb    tmp3, [dst]
     pop     {result, tmp1, tmp2, tmp3, src_offset}
+#if __ARM_FEATURE_PAC_DEFAULT
+	aut ip, lr, sp
```
Comment (xiaoxiang781216) [inline libs/libc/machine/arm/armv8-m/gnu/arch_strcpy.S:228]: align

File: libs/libc/machine/arm/armv8-m/gnu/arch_strcpy.S [hunk 4]
```diff
@@ -63,6 +71,13 @@
     .type   strcpy, %function
 
 strcpy:
+#if __ARM_FEATURE_PAC_DEFAULT
+#  if __ARM_FEATURE_BTI_DEFAULT
+	pacbti  ip, lr, sp
+#  else
+	pac ip, lr, sp
```
Comment (xiaoxiang781216) [inline libs/libc/machine/arm/armv8-m/gnu/arch_strcpy.S:78]: let's use the indent style

Comment (anchao) [reply libs/libc/machine/arm/armv8-m/gnu/arch_strcpy.S]: Done

File: libs/libc/machine/arm/armv8-m/gnu/arch_strcpy.S (hunk 3 above)
Comment (anchao) [reply libs/libc/machine/arm/armv8-m/gnu/arch_strcpy.S]: Done

File: libs/libc/machine/arm/armv8-m/gnu/arch_strlen.S (hunk 2 above)
Comment (anchao) [reply libs/libc/machine/arm/armv8-m/gnu/arch_strlen.S]: Done

File: libs/libc/machine/arm/armv8-m/gnu/arch_strlen.S (hunk 1 above)
Comment (anchao) [reply libs/libc/machine/arm/armv8-m/gnu/arch_strlen.S]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**